### PR TITLE
Fix file share with caption quote reply

### LINKED ISSUE #11108: Quotes of files with caption are too big
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Upload a file with a caption that has multiple lines
2. Quote that file with captions

### Expected behaviour
The caption is shortened, as with normal chat messages

### Actual behaviour
image

Also the "->" is escaped in the quote, while this does not happen on normal chat messages.

## Talk app

**Talk app version:** 18 RC2 and main

Tested with Chrome

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Fix #11108

### 🖼️ Screenshots / Screencasts

| 🏚️ Before | 🏡 After |
|------------|----------|
| image   | image      |

### 🚧 Tasks

- [ ] Code review

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [x] 🖌️ Design was reviewed, approved or inspired by the design team
- [x] ⛑️ Tests are included or not possible
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/NewMessage/NewMessage.vue [hunk 1]
```diff
@@ -506,6 +506,12 @@ export default {
 				}
 			}
 
+			// FIXME upstream: https://github.com/nextcloud-libraries/nextcloud-vue/issues/4492
+			const temp = document.createElement('textarea')
+			temp.innerHTML = this.text.replace(/&/gmi, '&amp;')
```
Comment (Antreesy) [inline src/components/NewMessage/NewMessage.vue:511]: Not sure if text couldn't be empty or undefined here

File: src/components/Quote.vue [hunk 2]
```diff
@@ -326,4 +331,7 @@ export default {
 	}
 }
 
+.file-preview {
```
Comment (Antreesy) [inline src/components/Quote.vue:334]: Was not sure about purpose of it. But if I got it right, it's not reducing the size of `<a>` tag
image

File: src/components/NewMessage/NewMessage.vue (hunk 1 above)
Comment (DorraJaouad) [reply src/components/NewMessage/NewMessage.vue:511]: it shouldn't be undefined in any case but we can add a condition to be not empty ( '' )

File: src/components/NewMessage/NewMessage.vue [hunk 3]
```diff
@@ -507,10 +507,12 @@ export default {
 			}
 
 			// FIXME upstream: https://github.com/nextcloud-libraries/nextcloud-vue/issues/4492
-			const temp = document.createElement('textarea')
-			temp.innerHTML = this.text.replace(/&/gmi, '&amp;')
-			this.text = temp.value.replace(/&amp;/gmi, '&').replace(/&lt;/gmi, '<')
-				.replace(/&gt;/gmi, '>').replace(/&sect;/gmi, '§')
+			if (this.text !== '') {
```
Comment (Antreesy) [inline src/components/NewMessage/NewMessage.vue:510]:
```suggestion
			if (this.hasText) {
```
it's trimmed in addition

File: src/components/Quote.vue [hunk 4]
```diff
@@ -329,9 +329,9 @@ export default {
 		position: relative;
 		margin: auto;
 	}
+	&__file__preview {
```
Comment (Antreesy) [inline src/components/Quote.vue:332]:
```suggestion
	&__file-preview {
```

File: src/components/Quote.vue [hunk 5]
```diff
@@ -41,13 +41,14 @@ components.
...
-					:arguments="richParameters"
-					:autolink="true" />
-			</div>
-			<blockquote v-else
+			<!-- file preview-->
+			<NcRichText v-if="isFileShareMessage"
+				text="{file}"
+				class="quote__file__preview"
+				:arguments="richParameters"
+				:autolink="true" />
```
Comment (Antreesy) [inline src/components/Quote.vue:49]: Do we need `autolink`, if it's file only?

**Review by Antreesy — APPROVED** [review]
Nothing blocking, nicely done. Left some nitpicks:

File: src/components/Quote.vue (hunk 5 above)
Comment (DorraJaouad) [reply src/components/Quote.vue:49]: forgot to remove it

Comment (nickvergessen) [reply src/components/Quote.vue:49]: and captions with links?

File: src/components/NewMessage/NewMessage.vue [hunk 6]
```diff
@@ -506,6 +506,14 @@ export default {
 				}
 			}
 
+			// FIXME upstream: https://github.com/nextcloud-libraries/nextcloud-vue/issues/4492
+			if (this.text.hasText) {
```
Comment (nickvergessen) [inline src/components/NewMessage/NewMessage.vue:510]: Old if was `if (this.hasText) {`
Is this intended to change?

**Review by Antreesy — APPROVED** [review]
Tested again

**Pull request merged**